cpp/CMakeLists.txt

```diff
@@ -116,15 +133,28 @@ if(DETECT_CONDA_ENV)
 endif()
 
 # ##################################################################################################
-# * compiler options ---------------------------------------------------------
+# * compiler options ----------------------------------------------------------
 
 set(_ctk_static_suffix "")
 if(CUDA_STATIC_RUNTIME)
   set(_ctk_static_suffix "_static")
 endif()
 
-# CUDA runtime
-rapids_cuda_init_runtime(USE_STATIC ${CUDA_STATIC_RUNTIME})
+if(NOT BUILD_CPU_ONLY)
+  # CUDA runtime
+  rapids_cuda_init_runtime(USE_STATIC ${CUDA_STATIC_RUNTIME})
+  # * find CUDAToolkit package
+  # * determine GPU architectures
+  # * enable the CMake CUDA language
+  # * set other CUDA compilation flags
+  rapids_find_package(
+    CUDAToolkit REQUIRED
+    BUILD_EXPORT_SET raft-exports
+    INSTALL_EXPORT_SET raft-exports
+  )
+else()
+  add_definitions(-DBUILD_CPU_ONLY)
```

Comment:
What does this line do? Does it add a `BUILD_CPU_ONLY` macro to every target?

Reply:
yes it does which I've used for quite some time, though would be good to change it to use https://cmake.org/cmake/help/latest/command/add_compile_definitions.html#command:add_compile_definitions which has superseded it

Reply:
This is the way I know how to do it, if it has superseded the other method then let's use it!